File: Python/src/sorts/quick_sort.py
def sort(array, low, high):
    if low >= high:
        return
    pivot = parition(array, low, high)
    sort(array, low, pivot - 1)
    sort(array, pivot + 1, high)

def parition(array, low, high):
    pivot = array[high]
    i = low - 1
    for j in range(low, high):
        if array[j] < pivot:
            i += 1
            array[i], array[j] = array[j], array[i]
    array[i + 1], array[high] = array[high], array[i + 1]
    return i + 1

File: CSharp/src/Sorts/QuickSort.cs
namespace Algorithm.Sorts;

public class QuickSort
{
    public void Sort(int[] array, int low, int high)
    {
        if (low >= high)
            return;
        
        var pivot = Partition(array, low, high);
        Sort(array, low, pivot - 1);
        Sort(array, pivot + 1, high);
    }

    private int Partition(int[] array, int low, int high)
    {
        var pivot = array[high];
        var i = low - 1;

        for (var j = low; j < high; j++)
        {
            if (array[j] < pivot)
            {
                i++;
                (array[i], array[j]) = (array[j], array[i]);
            }
        }
        (array[i + 1], array[high]) = (array[high], array[i + 1]);
        return i + 1;
    }
}